presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPruneCountAggregationOverScalar.java

```diff
@@ -13,47 +13,24 @@
  */
 package com.facebook.presto.sql.planner.iterative.rule;
 
-import com.facebook.presto.connector.ConnectorId;
-import com.facebook.presto.metadata.TableHandle;
 import com.facebook.presto.spi.type.BigintType;
-import com.facebook.presto.sql.planner.Symbol;
 import com.facebook.presto.sql.planner.iterative.rule.test.BaseRuleTest;
+import com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder;
+import com.facebook.presto.sql.planner.iterative.trait.CardinalityTrait;
 import com.facebook.presto.sql.planner.plan.AggregationNode;
-import com.facebook.presto.sql.planner.plan.Assignments;
 import com.facebook.presto.sql.tree.FunctionCall;
 import com.facebook.presto.sql.tree.QualifiedName;
-import com.facebook.presto.sql.tree.SymbolReference;
-import com.facebook.presto.tpch.TpchColumnHandle;
-import com.facebook.presto.tpch.TpchTableHandle;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import org.testng.annotations.Test;
 
-import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
 import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
-import static com.facebook.presto.tpch.TpchMetadata.TINY_SCALE_FACTOR;
 
 public class TestPruneCountAggregationOverScalar
         extends BaseRuleTest
 {
     @Test
-    public void testDoesNotFireOnNonNestedAggregate()
-    {
-        tester().assertThat(new PruneCountAggregationOverScalar())
-                .on(p ->
-                        p.aggregation((a) -> a
-                                .globalGrouping()
-                                .addAggregation(
-                                        p.symbol("count_1", BigintType.BIGINT),
-                                        new FunctionCall(QualifiedName.of("count"), ImmutableList.of()),
-                                        ImmutableList.of(BigintType.BIGINT))
-                                .source(
-                                        p.tableScan(ImmutableList.of(), ImmutableMap.of())))
-                ).doesNotFire();
-    }
-
-    @Test
-    public void testFiresOnNestedCountAggregate()
+    public void testDoesNotFireOnNestedCountAggregate()
```

Comment:
@kokosing I assumed that using traits in `PruneCountAggregationOverScalar` won't change the existing behavior. Why did the tests change?

Reply:
Because calculation of the cardinality was part of the rule (as it was using `CardinalityQueryUtil`), removed tests were partially testing `CardinalityQueryUtil` not the rule itself. I am not sure but it might be possible that previously rule didn't use `CardinalityQueryUtil` but something custom and then it was refactored to use `CardinalityQueryUtil`.

Reply:
@kokosing I'm not sure I understand. If the behavior changed, then it suggests that `QueryCardinalityUtil.isScalar` is not equivalent to checking cardinality trait. Is this because `QueryCardinalityUtil.isScalar` didn't support some of the cases that are now supported via traits ?

Reply:
Please notice that in this this tests there are no cardinality traits rules used. So no cardinality trait is known here and so rule is not able to rewrite the plan. Previously determining the cardinality was part of this rule, now it is a trait "dependency". 

Maybe I should just remove these tests? It seems that they are misleading. Now rules does not care about the shape of aggregation source but only traits, tests for different aggregation source plans suggest otherwise.

Please `testFireCountAggregateOverPlanWithExactCardinality` where cardinality injected into test plan is injected.